Essentials/src/main/java/com/earth2me/essentials/utils/VersionUtil.java

```diff
@@ -28,12 +30,46 @@
     public static final BukkitVersion v1_15_R01 = BukkitVersion.fromString("1.15-R0.1-SNAPSHOT");
     public static final BukkitVersion v1_15_2_R01 = BukkitVersion.fromString("1.15.2-R0.1-SNAPSHOT");
     public static final BukkitVersion v1_16_1_R01 = BukkitVersion.fromString("1.16.1-R0.1-SNAPSHOT");
-    public static final BukkitVersion v1_16_2_R01 = BukkitVersion.fromString("1.16.2-R0.1-SNAPSHOT");
     public static final BukkitVersion v1_16_4_R01 = BukkitVersion.fromString("1.16.4-R0.1-SNAPSHOT");
 
     private static final Set<BukkitVersion> supportedVersions = ImmutableSet.of(v1_8_8_R01, v1_9_4_R01, v1_10_2_R01, v1_11_2_R01, v1_12_2_R01, v1_13_2_R01, v1_14_4_R01, v1_15_2_R01, v1_16_4_R01);
 
+    private static final Map<String, SupportStatus> unsupportedServerClasses;
+
+    static {
+        final ImmutableMap.Builder<String, SupportStatus> builder = new ImmutableMap.Builder<>();
+
+        // Yatopia - Extremely volatile patch set;
+        //   * Messes with bungeecord UUIDs
+        //   * Frequent data corruptions
+        builder.put("org.yatopiamc.yatopia.server.YatopiaConfig", SupportStatus.DANGEROUS_FORK);
+
+        // KibblePatcher - Dangerous bytecode editor snakeoil that's only use is to break plugins
+        builder.put("net.kibblelands.server.FastMath", SupportStatus.DANGEROUS_FORK);
+
+        // AirplaneLite - Yatopia sidestream;
+        //   * Attempts unsafe chunk concurrency
+        builder.put("gg.airplane.structs.ChunkMapMap", SupportStatus.DANGEROUS_FORK);
+        builder.put("gg.airplane.structs.ConcLong2ObjectOpenHashMap", SupportStatus.DANGEROUS_FORK);
+
+        // Akarin - Dangerous patch history;
+        //   * Potentially unsafe saving of nms.JsonList
+        builder.put("io.akarin.server.Config", SupportStatus.DANGEROUS_FORK);
+
+        // Forge - Doesn't support bukkit
+        builder.put("net.minecraftforge.common.MinecraftForge", SupportStatus.UNSTABLE);
+
+        // Fabric - Doesn't support bukkit
+        builder.put("net.fabricmc.loader.launch.knot.KnotServer", SupportStatus.UNSTABLE);
+
+        // Misc translation layers that do not add NMS will be caught by this
+        builder.put("!net.minecraft.server." + ReflUtil.getNMSVersion() + ".MinecraftServer", SupportStatus.NMS_CLEANROOM);
```

Comment:
iirc calling this will break startup entirely on Glowstone or other non-NMS servers, which isn't ideal

Reply:
fixed this, will not silently fail